Source/ROSIntegration/Private/Conversion/Messages/sensor_msgs/SensorMsgsImageConverter.cpp

```diff
@@ -32,7 +32,8 @@ bool USensorMsgsImageConverter::ConvertOutgoingMessage(TSharedPtr<FROSBaseMsg> B
 	"encoding", BCON_UTF8(TCHAR_TO_UTF8(*Image->encoding)),
 	"step", BCON_INT32(Image->step)
 	);
-	bson_append_binary(*message, "data", -1, BSON_SUBTYPE_BINARY, Image->data, Image->height * Image->step );
-	// bson_append_binary(*message, "data", -1, BSON_SUBTYPE_BINARY, Image->data, 512 * 512 );
+	CSV_SCOPED_TIMING_STAT_GLOBAL(ConvertingImageMsg);
```

Comment:
If you are going to leave this in the code, wouldn't it be better to at least wrap it with an `#ifdef` guard to save cpu time when not debugging?

Reply:
I don`t think so, because I already control its usge using the UE4Editor commandline
>Simple threaded CSV Profiler To capture a CSV: - On the commandline, add -csvCaptureFrames=N to capture N frames from startup - On the console, use csvprofile start, csvprofile stop or csvprofile frames=N to capture a fixed number of frames - Instrument with CSV_SCOPED_STAT(statname), CSV_CUSTOM_STAT(statname,value). CSV capture is enabled in all builds except shipping. This change does not include any instrumentation of code, although basic unit stats are output. [Source](https://docs.unrealengine.com/en-US/Support/Builds/ReleaseNotes/4_17/index.html)

As far as I have seen the definitions:
```cpp
void FCsvProfiler::BeginFrame()
```
and
```cpp
void FCsvProfiler::EndFrame()
```
I don`t know if it makes sense control it also with a #ifdef flag.

Reply:
No, always development, I agree that the symbols will be there apart from shipping, but the instrumentation will be doing nothing, meaning that will not slow the execution, this is what I understood from the code

Reply:
> but the instrumentation will be doing nothing

Are you sure? If that's the case then my point is moot..